File: src/main/java/com/dv/telegram/command/ListSettings.java
package com.dv.telegram.command;

import com.dv.telegram.WikiBot;
import com.dv.telegram.config.BotSetting;
import org.apache.commons.lang3.StringUtils;

import java.util.ArrayList;
import java.util.List;

public class ListSettings extends BasicBotCommand {

    @Override
    public String getName() {
        return ListSettings.class.getSimpleName();
    }

    @Override
    public String getDescription(WikiBot bot) {
        return String.format(
            "`%s %s` — вывести список изменяемых в рантайме настроек бота.",
            bot.getBotName(),
            getCommandText()
        );
    }

    @Override
    public boolean useMarkdownInResponse() {
        return true;
    }

    @Override
    public String getDefaultCommandName() {
        return "/listSettings";
    }

    @Override
    public String getResponse(String text, WikiBot bot) {
        List<String> settingsLines = new ArrayList<>();

        for (BotSetting<?> setting : bot.getSettings().settings.values()) {
            settingsLines.add(String.format(
                "— *%s*:%n%s",
                setting.getName(),
                getSettingValueForMarkdown(setting)
            ));
        }

        return StringUtils.join(settingsLines, "\n\n");
    }
}

File: src/main/kotlin/com/dv/telegram/command/ListSettings.kt
package com.dv.telegram.command

import com.dv.telegram.WikiBot
import org.telegram.telegrambots.meta.api.objects.Update

class ListSettings : BasicBotCommand() {
    override val name: String = javaClass.simpleName

    override fun getDescription(bot: WikiBot) =
        "`${bot.botName} $commandText` — вывести список изменяемых в рантайме настроек бота."

    override fun useMarkdownInResponse() = true

    override val defaultCommandName = "/listSettings"

    override fun getResponse(text: String, bot: WikiBot, update: Update): String {
        val settingsLines = mutableListOf<String>()

        for (setting in bot.settings.settings.values) {
            settingsLines.add(
                "— *${setting.name}*:\n${getSettingValueForMarkdown(setting)}",
            )
        }

        return settingsLines.joinToString("\n\n")
    }
}
